cluster/validation.go

```diff
@@ -731,3 +738,19 @@ func getClusterVersion(version string) (semver.Version, error) {
 	}
 	return parsedVersion, nil
 }
+
+// warnWeaveDeprecation prints a deprecation warning if version is higher than 1.27
+func warnWeaveDeprecation(k8sVersion string) error {
+	version, err := util.StrToSemVer(k8sVersion)
+	if err != nil {
+		return fmt.Errorf("failed to translate: %s to semver notation: %w", k8sVersion, err)
+	}
+	version127, err := util.StrToSemVer("v1.27.0")
+	if err != nil {
+		return fmt.Errorf("failed to tranlate v1.27.0 to semver notation: %w", err)
+	}
+	if !version.LessThan(*version127) {
+		logrus.Warn("Weave CNI plugin is deprecated and will be removed in v1.30 kubernetes release")
```

Comment:
I would make this as extended as possible, something like:

```
Weave CNI plugin is deprecated starting with RKE v1.5.0 and Kubernetes v1.27.x and will be removed in Kubernetes v1.30.x
```

Or something along those lines.

Reply:
ok